pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
+	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+)
+
+func newPackagePublishCmd() *cobra.Command {
+
+	var err error
+	var path string
+	var tag string
+
+	cmd := &cobra.Command{
+		Use:   "publish-sdk <package> <language>",
+		Args:  cobra.ExactArgs(2),
+		Short: "Publish a package SDK to supported package registries.",
+		Long:  "Publish a package SDK to supported package registries.",
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
We don't generally use PreRun functions instead preferring everything in one Run function. That generally makes it easier to rip out and test a single function if needed.